dulwich/config.py

```diff
@@ -252,35 +252,52 @@ def has_section(self, name: Tuple[bytes, ...]) -> bool:
         return name in self.sections()
 
 
-class ConfigDict(Config, MutableMapping):
+BytesLike = Union[bytes, str]
+Key = Tuple[bytes, ...]
+KeyLike = Union[bytes, str, Tuple[BytesLike, ...]]
+Value = Union[bytes, bool]
```

Comment:
not sure whether a `bool` is really intended to be allowed here, but `set()` goes
```
if not isinstance(value, (bool, bytes)):
  ....
```
which suggests that it is.

Perhaps only bytes are intended, though, and that code in `set()` is redundant?